discovery_engine_core/core/src/stack/filters.rs

```diff
@@ -30,22 +33,58 @@ pub(crate) trait ArticleFilter {
     ) -> Result<Vec<Article>, GenericError>;
 }
 
+struct UniqueArticle(Article);
```

Comment:
This only make sense for DuplicateArticle so I'll prefer to keep it somewhere related to that. This should be used by other filters.

Reply:
I moved the definition into the function.